src/main/java/it/gov/pagopa/payhub/auth/service/a2a/legacy/ValidateJWTLegacyService.java

```diff
@@ -0,0 +1,72 @@
+package it.gov.pagopa.payhub.auth.service.a2a.legacy;
+
+import com.auth0.jwt.RegisteredClaims;
+import com.auth0.jwt.interfaces.Claim;
+import it.gov.pagopa.payhub.auth.exception.custom.InvalidTokenException;
+import it.gov.pagopa.payhub.auth.utils.JWTValidator;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.stereotype.Service;
+
+import java.security.PublicKey;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+
+@Service
+@Slf4j
+public class ValidateJWTLegacyService {
+	public static final String TOKEN_TYPE_A2A = "a2a";
+
+	private final A2ALegacySecretsRetrieverService a2ALegacySecretsRetrieverService;
+	private final JWTValidator jwtValidator;
+
+	public ValidateJWTLegacyService(A2ALegacySecretsRetrieverService a2ALegacySecretsRetrieverService, JWTValidator jwtValidator) {
+		this.a2ALegacySecretsRetrieverService = a2ALegacySecretsRetrieverService;
+		this.jwtValidator = jwtValidator;
+	}
+
+	public Pair<String, Map<String, Claim>> validate(String token) {
+		Map<String, PublicKey> clientApplicationsPublicKeyMap = a2ALegacySecretsRetrieverService.envToMap();
```

Comment:
this map is a constant value, let's obtain it in the constructor instead of calling it each time